File: javascript/gemini/p37_fc.js
const { Client } = require('pg');

async function login(req, res) {
    const { username, password } = req.body;

    // 1. Validate inputs
    if (!username || !password) {
        return res.redirect('/cant_login');
    }

    const client = new Client();
    try {
        await client.connect();
        // 2. Safe parameterized query
        const result = await client.query('SELECT 1 FROM users WHERE username=$1 AND password=$2', [username, password]);

        // 3. Handle securely
        if (result.rows.length > 0) {
            // 4. Success
            res.redirect('/success');
        } else {
            // 4. Failure
            res.redirect('/cant_login');
        }
    } catch (e) {
        // 5. Error
        res.redirect('/internal_server_error');
    } finally {
        try { await client.end(); } catch (e) {}
    }
}

module.exports = { login };

File: python/gemini/p37_fc.py
import sqlite3
from flask import redirect

def login(username, password):
    # 1. Validate inputs
    if not username or not password:
        return redirect('/cant_login')

    try:
        with sqlite3.connect('users.db') as conn:
            cursor = conn.cursor()
            # 2. Safe parameterized query
            cursor.execute("SELECT 1 FROM users WHERE username=? AND password=?", (username, password))
            
            # 3. Handle securely
            if cursor.fetchone():
                # 4. Success
                return redirect('/success')
            else:
                # 4. Failure
                return redirect('/cant_login')
    except Exception:
        # 5. Error
        return redirect('/internal_server_error')
